packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -214,35 +230,36 @@ export const Stackbox = () => {
             }}
           />
           {fromToken && balance && (
-            <div className="flex items-center justify-between">
+            <div className="flex flex-wrap items-center justify-between gap-3">
               <div className="flex space-x-1">
                 {balanceOptions.map(({ name, divider }) => (
                   <Button
                     key={name}
                     variant="secondary"
                     width="fit"
                     size="xs"
-                    onClick={() => {
-                      setShowTokenAmountError(false);
-                      setShowInsufficentBalanceError(false);
-                      setTokenAmount(
-                        formatUnits(
-                          balance.value / BigInt(divider),
-                          fromToken.decimals
-                        )
-                      );
-                    }}
+                    onClick={() => setTokenAmountBasedOnBalance(divider)}
                   >
                     {name}
                   </Button>
                 ))}
               </div>
               <div className="flex items-center space-x-1">
                 <TokenIcon token={fromToken} size="2xs" />
-                <BodyText className="text-em-high">
-                  <span className="text-em-low">Balance:</span>{" "}
-                  {formattedBalance(balance)}
-                </BodyText>
+                <div className="flex items-center space-x-1">
+                  <BodyText className="flex items-center text-em-low">
+                    Balance:
+                  </BodyText>
+                  <div
+                    onClick={() =>
+                      setTokenAmountBasedOnBalance(BalanceDivider.MAX)
+                    }
+                  >
```

Comment:
We shouldn't use divs with onClick, this should be a button.

Reply:
Also, we can declare the button inside the `BodyText` and just apply a `className="cursor-pointer hover:underline text-em-high"` to it.